### ISSUE TITLE
Ubuntu: Unable to get a font file. Please provide valid font file

### ISSUE THREAD

**Issue by tgerighty** [ehendrix23/tesla_dashcam#74]
Tesla_dashcam installed and working up to its help output. But then I try to run against my files I am getting the following error:

`Unable to get a font file. Please provide valid font file.`

What did I miss?

**Comment by ehendrix23** [comment]
What platform is this on? Windows, Linux, Mac? The error is that there is no default font for detected platform.

You can:
1. Use parameter --no-timestamp do not put the timestamp within the video.
2. Use parameter --font to provide path and filename for a valid font (ttf) file.

**Comment by tgerighty** [comment]
Ubuntu - I had put it in the issue title.

**Comment by ehendrix23** [comment]
Missed that. :-)

If you use with -h to get help, what does it provide as the default for parameter --font?

**Comment by tgerighty** [comment]
```
  --font FONT           Fully qualified filename (.ttf) to the font to be
                        chosen for timestamp. (default: None)
```

and 

```
 python3 -c 'import sys; print sys.platform'
  File "<string>", line 1
    import sys; print sys.platform
                        ^
SyntaxError: invalid syntax
```

**Comment by ehendrix23** [comment]
Use this then:
`python3 -c 'import sys; print(sys.platform)'`

Trying to see what you get there as a result. Normally would be expecting linux to be returned there.
Based on the default being None it clearly is that the platform returned is different then what we normally have.

**Comment by tgerighty** [comment]
```
python3 -c 'import sys; print(sys.platform)'
linux
```

**Comment by moorecp** [comment]
This also happens with the docker image:

    Provided font file /usr/share/fonts/truetype/freefont/FreeSans.ttf does exist, please provide a valid font file.

**Comment by ehendrix23** [comment]
> This also happens with the docker image:
> 
> ```
> Provided font file /usr/share/fonts/truetype/freefont/FreeSans.ttf does exist, please provide a valid font file.
> ```

There was an error in the comment, it should state:
`Provided font file /usr/share/fonts/truetype/freefont/FreeSans.ttf does not exist, please provide a valid font file.`

Does /usr/share/fonts/truetype/freefont/FreeSans.ttf exist?

**Comment by moorecp** [comment]
> Does  /usr/share/fonts/truetype/freefont/FreeSans.ttf exist?

Nope.

    root@03aeb730a589:/usr/src/app/tesla_dashcam# ls -R /usr/share/fonts/truetype/
    /usr/share/fonts/truetype/:
    dejavu

    /usr/share/fonts/truetype/dejavu:
    DejaVuSans-Bold.ttf  DejaVuSans.ttf  DejaVuSansMono-Bold.ttf  DejaVuSansMono.ttf  DejaVuSerif-Bold.ttf  DejaVuSerif.ttf
    root@03aeb730a589:/usr/src/app/tesla_dashcam#

Seems like that's not part of the `python:3` image. Looks like if you add `fonts-freefont-ttf` to the packages needing to be installed, it gets installed.

**Comment by ehendrix23** [comment]
> python3 -c 'import sys; print(sys.platform)'
> linux

Hmmm, this is weird. It is returning linux which is what I expect for Linux systems. When I test it by specifically stating linux it works, yet not working on yours.
Right now not sure why this is not working. 0.1.15 will have some more description (including platform) that might give a clue.

**Comment by ehendrix23** [comment]
Just released 0.1.15 with fix for missing fonts in Docker image.

@tgerighty, can you please run with this one. It does not contain a fix for the issue as currently not sure yet what the issue is. But the message being provided has been improved and hopefully might give me a better clue.

**Comment by tgerighty** [comment]
Not sure I got that version

```
sudo pip3 install tesla_dashcam --upgrade
Requirement already up-to-date: tesla_dashcam in /usr/local/lib/python3.6/dist-packages
Requirement already up-to-date: tzlocal in /usr/local/lib/python3.6/dist-packages (from tesla_dashcam)
Collecting pytz (from tzlocal->tesla_dashcam)
  Downloading https://files.pythonhosted.org/packages/87/76/46d697698a143e05f77bec5a526bf4e56a0be61d63425b68f4ba553b51f2/pytz-2019.2-py2.py3-none-any.whl (508kB)
    100% |████████████████████████████████| 512kB 720kB/s 
Installing collected packages: pytz
  Found existing installation: pytz 2018.3
    Not uninstalling pytz at /usr/lib/python3/dist-packages, outside environment /usr
Successfully installed pytz-2019.2
```

**Comment by c-E-c-777** [comment]
I was having the same issues as tgerighty and ended up resolving it by simply specifying 'python3.7' rather than just 'python3'. Your error clued me into it as it states it's using python3.6 whereas the project requires 3.7. Hope this helps.

**Comment by rounders** [comment]
```apt-get install ttf-freefont ``` solved this error for me on ubuntu 16.04
